rand_distr/tests/cdf.rs

```diff
@@ -159,6 +158,377 @@ fn normal() {
+        }
+        if k > n as i64 {
+            return 1.0;
+        }
+        gen_harmonic(k as u64, s) / gen_harmonic(n, s)
+    }
+
+    let parameters = [(1000, 1.0), (500, 2.0), (1000, 0.5)];
+
+    for (seed, (n, x)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Zipf::new(n, x).unwrap();
+        test_discrete(seed as u64, dist, |k| cdf(k, n, x));
+    }
+}
+
+#[test]
+fn gamma() {
+    fn cdf(x: f64, shape: f64, scale: f64) -> f64 {
+        if x < 0.0 {
+            return 0.0;
+        }
+
+        (x / scale).inc_gamma(shape)
+    }
+
+    let parameters = [
+        (0.5, 2.0),
+        (1.0, 1.0),
+        (10.0, 0.1),
+        (100.0, 0.0001),
+        (0.9999, 2.0),
+    ];
+
+    for (seed, (shape, scale)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Gamma::new(shape, scale).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, shape, scale));
+    }
+}
+
+#[test]
+fn chi_squared() {
+    fn cdf(x: f64, k: f64) -> f64 {
+        if x < 0.0 {
+            return 0.0;
+        }
+
+        (x / 2.0).inc_gamma(k / 2.0)
+    }
+
+    let parameters = [
+        // 0.01, // Fail case
+        0.1, 1.0, 2.0, 10.0, 100.0, 1000.0,
+    ];
+
+    for (seed, k) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::ChiSquared::new(k).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, k));
+    }
+}
+#[test]
+fn studend_t() {
+    fn cdf(x: f64, df: f64) -> f64 {
+        let h = df / (df + x.powi(2));
+        let ib = 0.5 * h.inc_beta(df / 2.0, 0.5, 0.5.ln_beta(df / 2.0));
+        if x < 0.0 {
+            ib
+        } else {
+            1.0 - ib
+        }
+    }
+
+    let parameters = [1.0, 10.0, 50.0];
+
+    for (seed, df) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::StudentT::new(df).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, df));
+    }
+}
+
+#[test]
+fn fisher_f() {
+    fn cdf(x: f64, m: f64, n: f64) -> f64 {
+        if (m == 1.0 && x <= 0.0) || x < 0.0 {
+            0.0
+        } else {
+            let k = m * x / (m * x + n);
+            let d1 = m / 2.0;
+            let d2 = n / 2.0;
+            k.inc_beta(d1, d2, d1.ln_beta(d2))
+        }
+    }
+
+    let parameters = [(1.0, 1.0), (1.0, 2.0), (2.0, 1.0), (50.0, 1.0)];
+
+    for (seed, (m, n)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::FisherF::new(m, n).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, m, n));
+    }
+}
+
+#[test]
+fn beta() {
+    fn cdf(x: f64, alpha: f64, beta: f64) -> f64 {
+        if x < 0.0 {
+            return 0.0;
+        }
+        if x > 1.0 {
+            return 1.0;
+        }
+        let ln_beta_ab = alpha.ln_beta(beta);
+        x.inc_beta(alpha, beta, ln_beta_ab)
+    }
+
+    let parameters = [
+        (0.5, 0.5),
+        (2.0, 3.5),
+        (10.0, 1.0),
+        (100.0, 50.0),
+        // (10.0, 0.1), // Fail case
```

Comment:
This is also a very extreme distribution concentrated around `1.0`
I will look into it in more detail later, for now we can drop this case.